docs/getting-started.md

```diff
@@ -42,9 +42,16 @@ Users of such distributions may have to install containerd from the source or a
 
 ##### systemd
 If you intend to start containerd via systemd, you should also download the `containerd.service` unit file from
-https://raw.githubusercontent.com/containerd/containerd/main/containerd.service into `/usr/local/lib/systemd/system/containerd.service`,
+`https://raw.githubusercontent.com/containerd/containerd/main/containerd.service` into `/usr/local/lib/systemd/system/containerd.service`,
+
+```console
+curl https://raw.githubusercontent.com/containerd/containerd/main/containerd.service | tee /usr/local/lib/systemd/system/containerd.service
```

Comment:
Why tee?

Reply:
```bash
mkdir -p /usr/local/lib/systemd/system
curl -SL -o /usr/local/lib/systemd/system/containerd.service https://raw.githubusercontent.com/containerd/containerd/main/containerd.service
```

Reply:
`tee` would probably be needed if `sudo` is required; `curl <url> | sudo tee <path>`

This should probably use `curl -fsSL` (or variations thereof)

Reply:
I wonder though if it would be better to describe how to use `systemctl` to edit/create services. I _think_ that would take care of creating the service file in the correct location;

```bash
systemctl edit myservice.service
No files found for myservice.service.
Run 'systemctl edit --force myservice.service' to create a new unit.
```

In the above, I try to edit a non-existing service; if the service already existed, it would open my editor to modify it, and if it doesn't exist, allow me to create a new one (which could be the content of the file that's in this repository)

Reply:
True, OTOH it also means the user is aware of what they're installing; `curl > containerd.service` means installing software that will run with full `root` permissions, which _should_ be something you'd want to verify before running.

Reply:
You make a good point on the security side; what's the general best-practice? For example I've seen commands that read from a remote (github) file and pipe directly to `sh`; may be provide a caution?
I can see how that could be dangerous, say if someone modified the url slightly to point to a compromised one e.g. `https://raw.githubusercontent.com/contanerd/containerd/main/containerd.service` (removed `i` in the org part that could be hard to spot at a quick glance).

Reply:
Yes, it's a tricky one. Even for the non-malicious case, this PR suggests using the `.service` file from the `main` branch, which may mean that changes introduced in `main` will now be installed by the user. The `main` branch may contain changes that are not meant for specific releases (_overall_ they should be fine, but it's possible). So from that perspective, using the `.service` file that matches the version of containerd you installed already may be a good thing to do at least. And even in that case _reviewing_ what the service looks like before installing is would be good practice.

`curl | sh` (or `curl | sudo sh`) is definitely "considered harmful". That won't _prevent_ users from doing so, but I think it's important to (when documenting) _describe_ best practices (which may not be the most "convenient"), and leave it to the user's creativity (and responsibility) to come up with alternatives (to whatever they're comfortable with, and what risk they're willing to take).

As an example, the "convenience" installation script for docker (at `get.docker.com`) contains this as comment at the top (and documentation has similar warnings);

```
# This script is meant for quick & easy install via:
#   $ curl -fsSL https://get.docker.com -o get-docker.sh
#   $ sh get-docker.sh
#
# For test builds (ie. release candidates):
#   $ curl -fsSL https://test.docker.com -o test-docker.sh
#   $ sh test-docker.sh
#
# NOTE: Make sure to verify the contents of the script
#       you downloaded matches the contents of install.sh
#       located at https://github.com/docker/docker-install
#       before executing.
```

(and yes, many users will disregard that warning and use `curl | sudo sh`)

Reply:
<details>
<summary> Let me know what you think about this rephrasing:</summary>
///<br/>
///<br/>
If you intend to start containerd via `systemd`, you should also download the `containerd.service` unit file from
`https://raw.githubusercontent.com/containerd/containerd/main/containerd.service` into `/usr/local/lib/systemd/system/containerd.service`,

```console
# create the path if not present
mkdir -p /usr/local/lib/systemd/system
# to verify the contents of the file as you write:
curl https://raw.githubusercontent.com/containerd/containerd/main/containerd.service | tee /usr/local/lib/systemd/system/containerd.service
# OR if you are sure about the contents, you can just
curl -SL -o /usr/local/lib/systemd/system/containerd.service https://raw.githubusercontent.com/containerd/containerd/main/containerd.service
```
</details>

@thaJeztah @AkihiroSuda

Reply:
@thaJeztah -- I really rather keep it simple and to the point to avoid the verbosity; it really goes along way improving the usability. Let's not get lost in the details. The current guide is good as it is, and I have followed it and I have a local one-node "cluster" running on my Ubuntu 22.04 VM (on Windows). First time setting this up.

Also another point, I'd really advocate for steps that can easily be scripted and run un-attended.

```
-vm:~$ k get nodes
NAME        STATUS   ROLES           AGE   VERSION
nandaa-vm   Ready    control-plane   10h   v1.26.0
```

Reply:
@thaJeztah @AkihiroSuda -- kindly let me know if I can revert to the initial simplified version of just recommending creation of the path if it doesn't exist; for now.